src/main/java/com/google/devtools/build/lib/analysis/RuleContext.java

```diff
@@ -190,6 +190,9 @@ public TransitiveInfoCollection getSrc() {
   /* lazily computed cache for Make variables, computed from the above. See get... method */
   private transient ConfigurationMakeVariableContext configurationMakeVariableContext = null;
 
+  /* cache for output artifacts, allows fetching them without creating/validating during Aspect processing */
+  private transient OutputArtifactsProvider outputArtifactsProvider;
```

Comment:
The `RuleContext` instance used during the analysis of a rule is thrown away after its analysis is finished and a fresh one is re-created when analyzing aspects attached to it. But then how come this works? I would expect that `createOutputArtifact()` is not called during the analysis of an aspect and thus this will always be empty.